src/main/java/shef/servlets/PreferenceServlet.java

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shef.servlets;
+
+import shef.data.User;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Arrays;
+import java.util.Map;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+
+/** Servlet to set the user's preferences. */
+@WebServlet("/preferences")
+public class PreferenceServlet extends HttpServlet {
+
+  /** This is a mapping of responses on the quiz to a list of preferences.
+    * For example, if the user selected "vegetarian" on the quiz, we must
+    * add all of the preferences defined here. (MEAT-FREE, VEGETARIAN, etc.)
+    */
+  public Map<String, List<String>> preferenceMap = new HashMap<String, List<String>>() {
+	  {
+      // Question 1: What type of person are you?
```

Comment:
I am not sure that this question is correct to ask users. Perhaps something like:
Which of the following describes (or may be some other word) you the most?

Reply:
I see that you have edited this is the html. Please edit here too.